### ISSUE TITLE
Portability issue: _PTHREAD_H does not exist

### ISSUE THREAD

**Issue by marcespie** [dtschump/gmic#317]
Every OS is free to have whatever convention they want for included header files. For instance, OpenBSD defines _PTHREAD_H_   for #include <pthread.h> 
 (note the extra underscore).
 
 
 I suggest using an API visible constant like PTHREAD_THREADS_MAX instead.

**Comment by marcespie** [comment]
`_PTHREAD_H_` of course... I forgot about the markup

**Comment by dtschump** [comment]
Does this commit https://github.com/dtschump/gmic/commit/8e85f02cf99eda19f195e6f58a14c46b82b4f6a6 solve the problem ?

**Comment by marcespie** [comment]
it should, but why not drop _PTHREAD_H entirely ?
I'm pretty sure PTHREAD_THREADS_MAX will be defined everywhere.

**Comment by dtschump** [comment]
> it should, but why not drop _PTHREAD_H entirely ?
> I'm pretty sure PTHREAD_THREADS_MAX will be defined everywhere.

I don't want to replace something that works (at least on many systems) by something that *may* work, unless there is a clear evidence that it would actually work.
When I read : https://pubs.opengroup.org/onlinepubs/7908799/xsh/pthread.h.html , there is no mention of the macro `PTHREAD_THREADS_MAX`. How can we be sure this macro exists for all implementations of pthread ?

I'm still investigating, but it may be even safer to consider another macro, just as one listed in the pthread page. Could you check for instance, if `PTHREAD_CANCELED` exists on BSD ?

**Comment by dtschump** [comment]
It looks like I'm actually using the `PTHREAD_CANCEL_ENABLE` macro in CImg.h, so this one should be present everywhere as it wouldn't compile otherwise.
I guess I'll take this one for the test then.

**Comment by dtschump** [comment]
https://github.com/dtschump/gmic/commit/e9cc5f8bd8091d7fa43342799fb2bd7cc01e395b

**Comment by dtschump** [comment]
May I close this issue?

**Comment by marcespie** [comment]
Yes, thank you !  I've updated gmic to the new release with the fix ;)
